numpy/lib/arraypad.py

```diff
@@ -74,6 +74,18 @@ def _round_ifneeded(arr, dtype):
         arr.round(out=arr)
 
 
+def _begin_slice(shape, n, axis):
+    """ Construct a slice tuple to take the first n element along axis """
+    return tuple(slice(None) if i != axis else slice(0, n)
+                 for (i, x) in enumerate(shape))
+
+
+def _end_slice(shape, n, axis):
```

Comment:
One doesn't actually need the `shape` for either `_begin_slice` (obvious, it is not used), or `_end_slice`: just do
```
def _end_slice(ndim, n, axis):
    return tuple(slice(None) if i != axis else slice(-n, None) for i in range(ndim))
```

Reply:
`-n` fails when `n == 0`

Reply:
Actually, thinking more, we do not need `ndim` either, since you only need a tuple up to `axis` - all the later dimensions are automatically taken to be `slice(None)`. So, it could be:
```
def _slice_at_axis(axis, slice):
    return (slice(None),) * axis + (slice,)
```
(This is substantially faster than the loop, at least for a randomly selected `axis=3`.)

Reply:
Perhaps pass in `slice(arr.shape[axis] - padding, None)` in that case - at least it is immediately obvious what the code does.

Reply:
I was trying to avoid writing `axis` twice, hence the lambda function above.

> This is substantially faster than the loop

I suspect this would be true if I kept my existing function names but changed their implementations too (I deliberately didn't do that in this PR, because it makes the diff clearer)